src/main/java/com/openshift/client/utils/TarFileUtils.java

```diff
@@ -35,7 +35,7 @@ public static boolean hasGitFolder(InputStream inputStream) throws IOException {
 		try {
 			boolean gitFolderPresent = false;
 			tarInputStream = new TarArchiveInputStream(new GzipCompressorInputStream(inputStream));
-			for (TarArchiveEntry entry = null; (entry = tarInputStream.getNextTarEntry()) != null;) {
```

Comment:
yes but if you change the impl to something that would not always set it and forget about changing the initialization....
It's simply very good but simple practice: always initialize local variables, instance variable are automagically always in a sane state. I really dont see the point of removing the existing initializations. Spare some characters? And as said, it's a good practice to always initialize local variables to a defined state.